src/audio/module_adapter/module/waves/waves.c

```diff
@@ -565,6 +565,44 @@ static int waves_effect_handle_param_message(struct processing_module *mod,
 	return ret;
 }
 
+static int waves_effect_verify_payload_size(struct processing_module *mod)
+{
+	int ret = 0;
+	struct comp_dev *dev = mod->dev;
+	struct module_data *codec = &mod->priv;
+	struct module_param *param;
+	struct module_config *cfg = &codec->cfg;
+
+	if (cfg->size > MAX_CONFIG_SIZE_BYTES) {
+		comp_err(dev, "waves_effect_verify_payload_size() provided config is too big, size %d",
+			cfg->size);
+		return -EINVAL;
+	}
+
+	/* sizes of individual module_param's must add up to 'cfg->size' */
+	int index = 0;
+	const uint32_t header_size = sizeof(param->size) + sizeof(param->id);
+
+	while (index < cfg->size && (!ret)) {
```

Comment:
Please remove `(!ret)` - you always terminate the loop if you set `ret`
Also could do:
```
for (index = 0; index < cfg->size; index += param->size)
```